### ISSUE TITLE
With 'smudge --skip', if the file exists in cache use it?

### ISSUE THREAD

**Issue by ajohnson23** [github/git-lfs#827]
TLDR: I like running with --skip by default to prevent the slow serial downloads, but it's inconvenient to have to always run 'git lfs pull' even when the objects are already cached locally. Please consider either changing the behavior of --skip or adding another flag that avoids *downloads* during smudge (outputting the placeholder in this case), but still smudges if the object is available in cache.

Long version:
Currently, with smudge --skip, you'll still get the lfs placeholder file even if you already had the file in your .git/lfs/objects cache.

It's mainly the *download* that hurts me in terms of wait time, so I'd prefer if there were a way to skip just downloading during smudge, but go ahead and do the smudge if the file already exists in the .git/lfs/objects cache. This way, if i do a 'git checkout tree-ish' and the lfs objects are already cached locally, they'll be smudged in. For files that aren't available in cache, I should still get a placeholder.

As far as I'm concerned the behavior of smudge --skip could just be changed to work this way, but if there's some reason for having a way to totally disable smudging (even when the file is cached), another flag such as --skip-downloads could be added.

**Comment by ajohnson23** [comment]
I opened a pull request for this, #831

**Comment by ajohnson23** [comment]
Merged a fix, closing this.
